sheets/2/exercise-5.tex

```diff
@@ -1,18 +1,49 @@
-    \subexcercise Schaue dir noch einmal den Code der vorherigen Aufgabe an.
-    Verändere ihn so, dass sowohl \lstinline{richNeo} als auch \lstinline{poorNeo} Attribute der Klasse \lstinline{Solution2_4} sind.
+\excercise{While i do stuff on repeat}
+%!TEX root = ./jvk-blatt2.tex
 
-    Verschiebe ausßerdem die Anweisungen zum Erzeugen der Objekte in die prepare-Operation.
+Ziel dieser Aufgabe ist es, die Syntax und die Funktion der While-Schleife genauer zu verstehen.
+\begin{enumerate}[label=\alph*)]
+    \item Instanziiere die Simulation wie bekannt und mache dich mit dieser vertraut.
+\end{enumerate}
+
+\begin{Infobox}[While Schleife]
+    Wir betrachten eine beispielhafte While Schleife
+    \begin{lstlisting}[breaklines=true, numbers=none]
+int i = 0;
+while(i < 3){
+    System.out.println(i);
+    i = i + 1;
+}
+    \end{lstlisting}
+    While Schleifen prüfen jeweils vor Ausführung des Schleifenrumpfes ob die gegebene Bedinung noch erfüllt ist. In unserem Beispiel 
+    wird also zunächst geprüft ob \textit{i \textless 3} mit dem Initialwert \textit{i = 0} gilt. Da dies erfüllt ist, wird der aktuelle Wert von \textit{i}
+    ausgegeben und inkrementiert. Nach der dritten Iteration ist der Wert von \textit{i} gleich 3, wodurch die obige Bedinung nicht mehr erfüllt ist
+    und die Schleife terminiert.\par 
+    Eine für uns relevantere Funktion wäre das Aufheben von einer unbekannten Anzahl an Münzen. Diese kann mittels einer While Schleife wie folgt 
+    realisiert werden:
+    \begin{lstlisting}[breaklines=true, numbers=none]
+Position coinPosition = //todo
+Iterator<Coin> collectableCoins = sim.getPlayfield().
+    getEntitiesOfTypeAt(coinPosition, Coin.class, true).
+    iterator();
+while(collectableCoins.hasNext()){
+    player.collect(collectableCoins.next());
+}
+    \end{lstlisting}
+    Dabei werden zunächst alle Münzen an einer bestimmten Position aufgezählt. Die Aufzählung selber findet mit Hilfe 
+    des Iterators statt, welche uns hier jetzt nicht weiter interessieren soll. Anschließend prüfen wir mit der Schleifenbedinung
+    jeweils, ob es noch weitere Münzen gibt die wir aufheben können und heben diese ggfs. in dem Schleifenrumpf auf.
+\end{Infobox}
+
+\begin{enumerate}[label=\alph*)] \setcounter{enumi}{1}
+    \item Betrachte die Methode \textit{run} in der Klasse \textit{Sheet2Task5}. Übernehme die obige While Schleife um 
+    alle Münzen auf dem Feld vor Neo aufzuheben.
+    \item Platziere alle Münzen auf dem rechten mittleren Spielfeld an der Wand. Implementiere das Verhalten in der \textit{run} Methode.
+    \item Zeichne mit den aufgehobenen Münzen eine Linie auf der mittleren Spur. Implementiere das Verhalten in der \textit{run} Methode so, dass 
+    auf jedem Feld der Spur genau eine Münze liegt.
+    \item Nun soll eine leicht kompliziertere Spur gezeichnet werden (weiterhin die mittlere Spur). Platziere auf dem ersten Spielfeld (von links gezählt)
```

Comment:
Mit einer while schleife sollten nur periodische muster gefragt werden. Zum zählen wird später noch die For-Schleife eingeführt.

Hier soll nur der Schleifenrumpf angepasst werden. z.B.: Droppe jedes zweite Feld eine Münze

```java
while(player.canDrop()) {
    player.drop();
    player.move();
    player.move();
}
```